data/generators.d.ts

```diff
@@ -0,0 +1,5 @@
+export type DataGenerator<T, O> = {
+    generate: (options?: O) => T
+}
+
+export type OptionsOrAll<T> = 'all' | T[]
```

Comment:
Would there be a value is extending this to include an explicit 'none' option to support when 'default' isn't just nothing but we might have tests cases for "when there is no data".

Add it onto the "to experiment with" pile